PukingPredator/Assets/Scripts/Consumable/Consumable.cs

```diff
@@ -1,353 +1,355 @@
+            targetRB.useGravity = enabled;
+        }
+    }
+
+    private void SetGravityDisabled()
+    {
+        SetGravity(false);
+    }
+
+    private void SetGravityEnabled()
+    {
+        SetGravity(true);
+    }
+    #endregion
+
+    /// <summary>
+    /// Moves the item to the position and reactivates it.
+    /// </summary>
+    /// <param name="position"></param>
+    public void PlaceAt(Vector3 position)
+    {
+        if (state != ItemState.inInventory) { return; }
+
+        gameObject.transform.localScale = initialScale;
+
+        var previousAngles = gameObject.transform.eulerAngles;
+        gameObject.transform.eulerAngles = new Vector3(0, previousAngles.y, 0);
+
+        gameObject.transform.position = position;
+        SetState(ItemState.inWorld);
+    }
+
+    private void ResetVelocity()
+    {
+        rb.velocity = Vector3.zero;
+        rb.angularVelocity = Vector3.zero;
+    }
+
+    #region Set Layer
+    /// <summary>
+    /// Set the layer of the instance and all of its children to the matching layer name.
+    /// </summary>
+    /// <param name="layer"></param>
+    private void SetLayer(int layer)
+    {
+        foreach (GameObject target in gameObject.GetDescendantsAndSelf())
+        {
+            target.layer = layer;
+        }
+    }
+
+    private void SetLayerToConsumable()
+    {
+        SetLayer(GameLayer.consumable);
+    }
+
+    private void SetLayerToConsumed()
+    {
+        SetLayer(GameLayer.consumed);
+    }
+    #endregion
+
+    /// <summary>
+    /// Swap between states
+    /// </summary>
+    /// <param name="state"></param>
+    public void SetState(ItemState newState)
+    {
+        var previousState = state;
+        if (newState == previousState) { return; }
+        state = newState;
+
+        stateEvents[previousState].onExit?.Invoke();
+        stateEvents[newState].onEnter?.Invoke();
+    }
+
+    /// <summary>
+    /// Can be called to make the instance start to decay.
+    /// </summary>
+    public void StartDecay()
+    {
+        if (!canDecay || isDecaying) { return; }
+
+        //if there is already a timer, it must be decaying already.
+        decayTimer = gameObject.AddComponent<Timer>();
+        decayTimer.StartTimer(decayTime);
+        decayTimer.onTimerComplete += Decay;
+    }
+
+    private void UpdateBeingConsumed()
+    {
+        SetRBKinematic(false);
+        var ownerPosition = ownerTransform.position;
+        var currRate = consumptionRate * Time.deltaTime;
+        gameObject.transform.position = Vector3.Lerp(gameObject.transform.position, ownerPosition, currRate);
+        gameObject.transform.localScale = Vector3.Lerp(gameObject.transform.localScale, Vector3.zero, currRate);
+
+        var hasBeenConsumed = gameObject.transform.localScale.magnitude / initialScale.magnitude < consumptionCutoff;
+        if (hasBeenConsumed) { SetState(ItemState.inInventory); }
+    }
+
+    private void UpdateProximityOutline()

```

Comment:
it can probably go, ill leave it for now though, doesnt do any harm other than taking up a few lines of code visually that likely gets optimized out of the compiled program anyways